src/main/java/com/coveo/spillway/AsyncLimitUsageStorage.java

```diff
@@ -0,0 +1,73 @@
+package com.coveo.spillway;
+
+import com.coveo.spillway.memory.InMemoryStorage;
+import com.coveo.spillway.memory.OverrideKeyRequest;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+public class AsyncLimitUsageStorage implements LimitUsageStorage {
+
+  private static final Logger logger = LoggerFactory.getLogger(AsyncLimitUsageStorage.class);
+
+  private final LimitUsageStorage wrappedLimitUsageStorage;
+  private final ExecutorService executorService;
+  private InMemoryStorage cache;
+
+  public AsyncLimitUsageStorage(LimitUsageStorage wrappedLimitUsageStorage) {
+    this.wrappedLimitUsageStorage = wrappedLimitUsageStorage;
+    this.executorService = Executors.newSingleThreadExecutor();
+    this.cache = new InMemoryStorage();
+  }
+
+  @Override
+  public Map<LimitKey, Integer> addAndGet(Collection<AddAndGetRequest> requests) {
+    Map<LimitKey, Integer> cachedEntries = cache.addAndGet(requests);
+    executorService.submit(() -> sendAndCacheRequests(requests));
```

Comment:
Might be better to `ScheduledExecutorService` to group together a series of request (with a pretty short delay, no more than a few seconds). This way you'd reduce the number of network calls made to Redis etc. Not sure if it's really worth it though, maybe I should measure before complaining :)

pour grouper une série de requests. Là tu gagnes en perf mais tu envoie encore une request à Re